src/main/app/src/pages/ScheduleActivityPage.js

```diff
@@ -205,7 +176,29 @@ const ScheduleActivityPage = ({isLoggedIn, accessToken, userId, activity, links,
           </Card>
         </div>
      :
-        generateRandomActivities(links)
+        <div className={classes.root} className="container">
+          {generateRandomActivities(links).map((element, key) => {
+            if (activityType === "games") {
+              return (
+                <div key={key}>
+                  <GameCard data={element} onClickFunction={alertUpdateActivity} parameters={element} buttonText={"Choose this activity"}/>
+                </div>
+              );
+            } else if (activityType === "reading") {
+              return (
+                <div key={key}>
+                  <ArticleCard data={element} onClickFunction={alertUpdateActivity} parameters={element} buttonText={"Choose this activity"}/>
+                </div>
+              );
+            } else if (activityType === "active") {
+              return (
+                <div key={key}>
```

Comment:
Also, since div key={key} line is same for all three cards. do you want to render it above everything and then just display the cards inside the if condition.

Reply:
I tried that, the issue is if I add it before the if function and returns, it doesn't read it as a function and instead reads it as html, causing the entire function to not be read.

Reply:
I think you could do: return ( [div key={key}] {acitivityType === "games" ? <GameCard ... /> : activityType === "reading" ? <ArticleCard ... /> : activityType === "active" ? <VideoCard ... /> : null } [/div] )

Reply:
I tried doing [div key={key}] before and it changes how the entire proceeding parts are read...

Reply:
Just added the changes! :)